de-lib/src/main/java/org/iplantc/de/client/models/QualifiedId.java

```diff
@@ -0,0 +1,44 @@
+package org.iplantc.de.client.models;
+
+/**
+ * @author dennis
+ */
+public class QualifiedId  implements HasQualifiedId {
+
+    private String systemId;
+    private String id;
+
+    @Override
+    public String getSystemId() {
+        return systemId;
+    }
+
+    @Override
+    public void setSystemId(String systemId) {
+        this.systemId = systemId;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public QualifiedId(final String systemId, final String id) {
+        this.systemId = systemId;
+        this.id = id;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof QualifiedId) {
+            QualifiedId other = (QualifiedId) obj;
+            return other.getSystemId() == systemId && other.getId() == id;
```

Comment:
Since these are strings, they should be compared with `.equals()` instead of `==`.